internal/provider/private_endpoint_services_resource.go

```diff
@@ -252,21 +253,23 @@ func loadEndpointServicesIntoTerraformState(
 			EndpointServiceId: types.StringValue(service.GetEndpointServiceId()),
 		}
 
-		if services[i].CloudProvider.String() == "AWS" {
-			services[i].Aws = PrivateLinkServiceAWSDetail{
-				ServiceName: types.StringValue(service.Aws.GetServiceName()),
-				ServiceId:   types.StringValue(service.Aws.GetServiceId()),
-			}
-		}
-
 		traceAPICall("GetAvailabilityZoneIds")
 		apiAZs := service.GetAvailabilityZoneIds()
 		azs := make([]types.String, len(apiAZs))
 		for j, az := range apiAZs {
 			azs[j] = types.StringValue(az)
 		}
 		services[i].AvailabilityZoneIds = azs
-		services[i].Aws.AvailabilityZoneIds = azs
+
+		// Set the .AWS field for backward compatibility.
+		// Note that `services[i].CloudProvider.String()` "\"AWS\"".
+		if strings.Contains(services[i].CloudProvider.String(), "AWS") {
```

Comment:
I think you can use `ValueString()` here to get the actual value inside.
```suggestion
		if services[i].CloudProvider.ValueString() == "AWS" {
```

Reply:
nice.